app/src/main/java/com/example/cactusnotes/SignUpActivity.kt

```diff
@@ -0,0 +1,36 @@
+package com.example.cactusnotes
+
+import android.os.Bundle
+import android.widget.Toast
+import androidx.appcompat.app.AppCompatActivity
+import com.example.cactusnotes.databinding.SignUpActivityBinding
+import com.example.cactusnotes.validations.EmailValidator
+import com.example.cactusnotes.validations.PasswordValidator
+import com.example.cactusnotes.validations.UsernameValidation
+
+
+class SignUpActivity : AppCompatActivity() {
+    private lateinit var binding: SignUpActivityBinding
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = SignUpActivityBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+        this.setTitle("Sign Up")
```

Comment:
1. User visible strings should always be extracted into strings.xml.
2. As far as I know the "recommended and widely used" way of setting title is `supportActionBar?.title = R.string.your_title`
3. If you decide to use Activity#setTitle use property access syntax instead and also drop the keyword "**this**": `title = R.string.your_title`